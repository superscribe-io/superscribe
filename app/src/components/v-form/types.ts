import { DeepPartial, Field } from '@directus/types';

export type FormField = DeepPartial<Field> & {
	field: string;
	name: string;
	hideLabel?: boolean;
	hideLoader?: boolean;
};
