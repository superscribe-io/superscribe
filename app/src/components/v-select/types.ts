export type Option = {
	value: string | number | null;
	icon?: string;
	text?: string;
	disabled?: boolean;
	children?: Option[];
	divider?: boolean;
	selectable?: boolean;
	hidden?: boolean;
};
