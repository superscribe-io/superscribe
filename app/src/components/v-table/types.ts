export type Alignment = 'left' | 'center' | 'right';

export type HeaderRaw = {
	text: string;
	value: string;
	description?: string | null;
	align?: Alignment;
	sortable?: boolean;
	width?: number | null;
	[key: string]: any;
};

export type Header = Required<HeaderRaw>;

export type Item = {
	[key: string]: any;
};

export type ItemSelectEvent = {
	value: boolean;
	item: Item;
};

export type Sort = {
	by: string | null;
	desc: boolean;
};
