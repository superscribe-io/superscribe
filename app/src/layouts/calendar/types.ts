export type LayoutOptions = {
	template?: string;
	startDateField?: string;
	endDateField?: string;
	firstDay?: number;
	viewInfo?: {
		type: string;
		startDateStr: string;
	};
};
