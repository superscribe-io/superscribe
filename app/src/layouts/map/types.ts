import { CameraOptions } from 'maplibre-gl';

export type LayoutQuery = {
	fields: string[];
	sort: string[];
	limit: number;
	page: number;
};

export type LayoutOptions = {
	cameraOptions?: CameraOptions & { bbox: any };
	geometryField?: string;
	autoLocationFilter?: boolean;
	clusterData?: boolean;
	animateOptions?: any;
	displayTemplate?: string;
};
