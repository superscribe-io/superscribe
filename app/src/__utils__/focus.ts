import { Directive } from 'vue';

export const Focus: Directive = {
	mounted(el, binding) {
		if (binding.value) {
			el.focus();
		} else {
			el.blur();
		}
	},
};
