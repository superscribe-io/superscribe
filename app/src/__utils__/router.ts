import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';

export function generateRouter(routes: RouteRecordRaw[]) {
	const router = createRouter({
		history: createWebHistory(),
		routes,
	});

	return router;
}
