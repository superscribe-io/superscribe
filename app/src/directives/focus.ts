import { Directive } from 'vue';

const Focus: Directive = {
	mounted(el, binding) {
		if (binding.value) {
			el.focus();
		} else {
			el.blur();
		}
	},
};

export default Focus;
