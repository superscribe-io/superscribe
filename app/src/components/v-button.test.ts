import { test, expect, beforeEach } from 'vitest';
import { mount } from '@vue/test-utils';

import VButton from './v-button.vue';
import { h } from 'vue';
import { generateRouter } from '@/__utils__/router';
import { Router } from 'vue-router';
import { GlobalMountOptions } from '@vue/test-utils/dist/types';
import { Focus } from '@/__utils__/focus';

let router: Router;

let global: GlobalMountOptions;

beforeEach(async () => {
	router = generateRouter([
		{
			path: '/',
			component: h('div', VButton),
		},
		{
			path: '/test',
			component: h('div', 'empty'),
		},
	]);

	router.push('/');
	await router.isReady();

	global = {
		stubs: ['v-progress-circular'],
		directives: {
			focus: Focus,
		},
		plugins: [router],
	};
});

test('Mount component', () => {
	expect(VButton).toBeTruthy();

	const wrapper = mount(VButton, {
		global,
	});

	expect(wrapper.html()).toMatchSnapshot();
});

// test('Click on link', async () => {
// 	const wrapper = mount(VButton, {
//         props: {
//             to: '/test'
//         },
// 		global
// 	});

//     await wrapper.get('a').trigger('click')

//     expect(router.currentRoute.value.path).toBe('/test')

// });
