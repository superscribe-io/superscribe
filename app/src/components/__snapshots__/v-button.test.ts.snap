// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Mount component 1`] = `
"<div data-v-28d526fe=\\"\\" class=\\"v-button\\"><button data-v-28d526fe=\\"\\" class=\\"button align-center normal\\" type=\\"button\\"><span data-v-28d526fe=\\"\\" class=\\"content\\"></span>
    <div data-v-28d526fe=\\"\\" class=\\"spinner\\">
      <!--v-if-->
    </div>
  </button></div>"
`;
